### ISSUE TITLE
Player Playing issue

### ISSUE THREAD

**Issue by gaganshrm23** [noreasonprojects/ModernAVPlayer#56]
Audio doesn't play sometime. It hang up between **Loading — Loaded — Buffering - Waiting For Network - Loading — Loaded — Buffering — Waiting For Network - Loading...** . Sometime audio played but takes too much time.

Logs : 

02:31:15076 [🔈][PlayerContext.swift]:85 :: Loading
2018-08-20 14:31:15.092040+0530 AudecibelPodcast[3935:1169149] CredStore - performQuery - Error copying matching creds.  Error=-25300, query={
    class = inet;
    "m_Limit" = "m_LimitAll";
    "r_Attributes" = 1;
    sync = syna;
}
2018-08-20 14:31:15.099437+0530 AudecibelPodcast[3935:1163830] [framework] CUICatalog: Invalid asset name supplied: ''

02:31:18381 [🔈][PlayerContext.swift]:85 :: Loaded
~~~ PLUGIN: customAttribute=attribute0
02:31:18382 [🔈][PlayerContext.swift]:85 :: Buffering
2018-08-20 14:31:18.387123+0530 AudecibelPodcast[3935:1163830] [framework] CUICatalog: Invalid asset name supplied: ''

02:31:21389 [🔈][PlayerContext.swift]:85 :: Waiting For Network
2018-08-20 14:31:21.396182+0530 AudecibelPodcast[3935:1163830] [framework] CUICatalog: Invalid asset name supplied: ''
2018-08-20 14:31:21.399674+0530 AudecibelPodcast[3935:1169965] Task <BD7D92BA-908D-41B6-AAD7-0C88881EAC01>.<1> finished with error - code: -999

02:31:21888 [🔈][PlayerContext.swift]:85 :: Loading
2018-08-20 14:31:21.905569+0530 AudecibelPodcast[3935:1169965] CredStore - performQuery - Error copying matching creds.  Error=-25300, query={
    class = inet;
    "m_Limit" = "m_LimitAll";
    "r_Attributes" = 1;
    sync = syna;
}
2018-08-20 14:31:21.915624+0530 AudecibelPodcast[3935:1163830] [framework] CUICatalog: Invalid asset name supplied: ''

02:31:24953 [🔈][PlayerContext.swift]:85 :: Loaded
~~~ PLUGIN: customAttribute=attribute0
02:31:24954 [🔈][PlayerContext.swift]:85 :: Buffering
2018-08-20 14:31:24.959167+0530 AudecibelPodcast[3935:1163830] [framework] CUICatalog: Invalid asset name supplied: ''

Please Help me solved this problem.
Thanks

**Comment by raphrel** [comment]
Do you have your own `PlayerConfiguration` implementation ? or use the default `ModernAVPlayerConfiguration`

**Comment by gaganshrm23** [comment]
I am using the default ModernAVPlayerConfiguration.

**Comment by raphrel** [comment]
so , first I see a image path problem. 
- localPlaceHolderImageName: the image is fetch from a bundle
- remoteImageUrl: the image will be download from a remote server
be sure any of the path are good. you can also remove from now the metadata to be sure it doesn't affect any audio problem

I probably have to change the way we inject image in metadata. (Swift Data type should be better)
-----
Also, waiting for network state load the url and check the request access to google.com (status 200), nothing more.
Your remote media file size is probably heavier than the google request.
So your network may be just good enough for the google request but not for fetching your media.

In the PlayerConfiguration, you can change the url: `reachabilityNetworkTestingURL`

**Comment by gaganshrm23** [comment]
I have never face image loading problem. The main issue is playing audio, I also noticed .mp4 format is not playing or it is taking too much time. Some Audio never played. Please check it once.

**Comment by raphrel** [comment]
ho sorry, I didn't see the links above.

I saw same problem here, so I will check as soon as possible what's wrong.

**Comment by gaganshrm23** [comment]
No Audio Duration for this URL : http://feedproxy.google.com/~r/PANPU_VOH/~5/5uNR2mVwTk4/PANPU_VOHX_20180819_1.mp3?#

**Comment by raphrel** [comment]
About the audio duration:

I tried with a simple AVPlayer project. use one asset with url mentionned: https://us1.hostedftp.com/~AWR-Ops/Podcasts/Audio/asia/PANPU/VOH/PANPU_VOHX_20180819_1.mp3

from the remote server:
duration KVO information is NAN. duration from AVPlayerItem / AVAsset is NAN too when item is readyToPlay.

I downloaded the file and import in the project:
no issue, duration is set correctly.

are you the owner of the mp3 file ? I suspect metadata mp3...
